### ISSUE TITLE
Create Custom Torch Dataset Class

### ISSUE THREAD

**Issue by avzero07** [avzero07/dl-based-sensor-calib#3]
To read in images and corresponding labels. Explore existing data-loader classes in pytorch.

**Comment by avzero07** [comment]
From #8 

- Updated gitignore to ignore `tag`
- Updated `deps/requirements.txt`
- Added `CarDataset` class. Map style dataset that derives from `torch.utils.data.Dataset`
- Added new tests `test/car_dataset_tests/test_dataset.py`
- Updated `test/car_dataset_tests/test_dataset.py` to sanitize frame name pattern

All new tests passing on Windows

<pre>(pytorch-cuda) PS C:\Users\aksha\Desktop\dl-based-sensor-calib> pytest -vv .\test\car_dataset_tests\test_dataset.py
============================================================================= test session starts =============================================================================
platform win32 -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1 -- D:\Development\Anaconda\envs\pytorch-cuda\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\aksha\Desktop\dl-based-sensor-calib
test/car_dataset_tests/test_dataset.py::test_check_dir[util-True] PASSED                                                                                                 [  3%]
test/car_dataset_tests/test_dataset.py::test_check_dir[test-True] PASSED                                                                                                 [  7%]
test/car_dataset_tests/test_dataset.py::test_check_dir[8nvMU-False] PASSED                                                                                               [ 11%]
test/car_dataset_tests/test_dataset.py::test_check_file[README.md-True] PASSED                                                                                           [ 14%]
test/car_dataset_tests/test_dataset.py::test_check_file[util\\car_dataset.py-True] PASSED                                                                                [ 18%]
test/car_dataset_tests/test_dataset.py::test_check_file[Ia09a-False] PASSED                                                                                              [ 22%]
test/car_dataset_tests/test_dataset.py::test_cardataset_init[folder_list0-label_file_list0-True] PASSED                                                                  [ 25%]
test/car_dataset_tests/test_dataset.py::test_cardataset_init[folder_list1-label_file_list1-False] PASSED                                                                 [ 29%]
test/car_dataset_tests/test_dataset.py::test_cardataset_init[folder_list2-label_file_list2-False] PASSED                                                                 [ 33%]
test/car_dataset_tests/test_dataset.py::test_cardataset_len PASSED                                                                                                       [ 37%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[0-True] PASSED                                                                                    [ 40%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[1199-True] PASSED                                                                                 [ 44%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[1200-True] PASSED                                                                                 [ 48%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[1201-True] PASSED                                                                                 [ 51%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[2399-True] PASSED                                                                                 [ 55%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[2400-False] PASSED                                                                                [ 59%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_index[-1-False] PASSED                                                                                  [ 62%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_angles[0-outcome0] PASSED                                                                         [ 66%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_angles[1199-outcome1] PASSED                                                                      [ 70%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_angles[1200-outcome2] PASSED                                                                      [ 74%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_angles[1201-outcome3] PASSED                                                                      [ 77%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_angles[2399-outcome4] PASSED                                                                      [ 81%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_images[0-0_Frames\\0_0001.jpg] PASSED                                                             [ 85%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_images[1199-0_Frames\\0_1200.jpg] PASSED                                                          [ 88%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_images[1200-3_Frames\\3_0001.jpg] PASSED                                                          [ 92%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_images[1201-3_Frames\\3_0002.jpg] PASSED                                                          [ 96%]
test/car_dataset_tests/test_dataset.py::test_cardataset_get_item_value_images[2399-3_Frames\\3_1200.jpg] PASSED                                                          [100%]

============================================================================== warnings summary ===============================================================================
D:\Development\Anaconda\envs\pytorch-cuda\lib\site-packages\pyreadline\py3k_compat.py:8
  D:\Development\Anaconda\envs\pytorch-cuda\lib\site-packages\pyreadline\py3k_compat.py:8: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    return isinstance(x, collections.Callable)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================================================================= 27 passed, 1 warning in 13.29s ========================================================================</pre>
